### DISCUSSION TITLE
Is there a way to make a plugin so GitExtensions supports UNIX paths (for wine use)?

### DISCUSSION THREAD

**Discussion by brunoais** [gitextensions/gitextensions#10184] [Q&A]
I'm still interested in trying to make GitExtensions runnable on wine but using the git executable which is available in the host system.
For that, I'm considering making a plugin that intercepts a path that looks "linux" enough and make it display and behave more friendly for linux users.

This would only modify a few parts:

1. User can input paths in linux format (they would be translated to drive Z:)
2. Every git command to a repository (on Z: drive) leads git extensions to run a specific script
3. Integration with ssh-agent of the parent system
4. Using linux native executable as linux tools (maybe possible?) Leaving this for later.

related to 2: the script calls a wine program which translates into an execution from the wine sandbox into unsandboxed linux's native git program with the native real path.
related to 3: I suppose that it should work out of the box due to how the ssh-agent is already in an env var. However, a closer look should help.

For now, no changes are to exist related to:

1. The paths displayed in the terminal execution screen (they stay with Z:\)
2. Miscellaneous display elements

Is this feasible? Does the plugin API allow such? If so, in what environment can I compile the plugin code on?

**Comment by gerhardol** [comment]
This would not be possible to do in a plugin, there are many internal paths that need to be handled.
[wsl-git ](https://github.com/andy-5/wslgit) implement this as a wrapper

In GE 4.0 an integrated approach is implemented with special handling where needed using "git" or "wsl git" where needed (for instance difftool is always "git").
"wine-git" could be done in a similar way, but in this case the wsl-git approach probably makes more sense. You could probably build on wsl-git. (The annoying with wsl-git for a Git user is that it unconditionally transforms regardless of the path by adding that the tool could be useful in many more situations.)

**Reply by brunoais** [reply]
I'll see if that works. Thanks.
